src/main/java/duke/storage/SaveParser.java

```diff
@@ -0,0 +1,170 @@
+        String[] taskType = this.scanner.nextLine().split(":", 2);
+        if (!taskType[0].equals("Task")) {
+            throw new DukeReadSaveException("Invalid Task");
+        }
+        switch (taskType[1]) {
+        case ("todo"):
+            return parseTodo();
+        case ("deadline"):
+            return parseDeadline();
+        case ("event"):
+            return parseEvent();
+        default:
+            throw new DukeReadSaveException("Invalid Task name");
+        }
+    }
+
+    private Todo parseTodo() throws DukeReadSaveException {
+        String name = "";
+        boolean isDone = false;
+        boolean[] argsFound = new boolean[2];
+
+        String nl;
+        while (this.scanner.hasNextLine()) {
+            if (this.scanner.hasNext("(Task:).*")) {
+                //break if next line is the start of a new task
+                break;
+            }
+            nl = this.scanner.nextLine();
+            if (nl.equals("")) {
+                break;
+            }
+            String[] key_value = nl.split(":", 2);
+            switch (key_value[0]) {
+            case ("\tName"):
+                name = key_value[1];
+                argsFound[0] = true;
+                break;
+            case ("\tDone"):
+                isDone = Boolean.parseBoolean(key_value[1]);
+                argsFound[1] = true;
+                break;
+            }
+        }
+        if (argsFound[0] && argsFound[1]) {
+            return new Todo(name, isDone);
+        }
+        throw MISSING_ARG_EXCEPTION;
+    }
+
+    private Deadline parseDeadline() throws DukeReadSaveException {
+        String name = "";
+        boolean isDone = false;
+        String by = "";
+        boolean[] argsFound = new boolean[3];
+
+        String nl;
+        while (this.scanner.hasNextLine()) {
+            if (this.scanner.hasNext("(Task:).*")) {
+                //break if next line is the start of a new task
+                break;
+            }
+            nl = this.scanner.nextLine();
+            if (nl.equals("")) {
+                break;
+            }
+            String[] key_value = nl.split(":", 2);
+            switch (key_value[0]) {
+            case ("\tName"):
+                name = key_value[1];
+                argsFound[0] = true;
+                break;
+            case ("\tDone"):
+                isDone = Boolean.parseBoolean(key_value[1]);
+                argsFound[1] = true;
+                break;
+            case ("\tBy"):
+                by = key_value[1];
+                argsFound[2] = true;
+                break;
+            }
+        }
+        if (argsFound[0] && argsFound[1] && argsFound[2]) {
+            return new Deadline(name, isDone, by);
+        }
+        throw MISSING_ARG_EXCEPTION;
+    }
+
+    private Event parseEvent() throws DukeReadSaveException {
+        String name = "";
+        boolean isDone = false;
+        String at = "";
+        boolean[] argsFound = new boolean[3];
```

Comment:
Perhaps you could try to avoid using "Magic Numbers" like 3 here, as it is not obvious what significance it has.

Reply:
Or maybe include a comment so other reviewers would know why 3 was used 😄

Reply:
Yup I agree some comments would be good for others to more easily follow the code! 😄